x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/StoreKeyConfig.java

```diff
@@ -82,43 +80,51 @@ X509ExtendedKeyManager createKeyManager(@Nullable Environment environment) {
     @Override
     X509ExtendedTrustManager createTrustManager(@Nullable Environment environment) {
         try {
-            return CertParsingUtils.trustManager(keyStorePath, keyStoreType, keyStorePassword.getChars(), trustStoreAlgorithm, environment);
-        } catch (Exception e) {
+            KeyStore ks = getStore(environment, keyStorePath, keyStoreType, keyStorePassword);
+            return CertParsingUtils.trustManager(ks, trustStoreAlgorithm);
+        } catch (IOException | CertificateException | NoSuchAlgorithmException | KeyStoreException e) {
             throw new ElasticsearchException("failed to initialize a TrustManagerFactory", e);
         }
     }
 
     @Override
     Collection<CertificateInfo> certificates(Environment environment) throws GeneralSecurityException, IOException {
-        final Path path = CertParsingUtils.resolvePath(keyStorePath, environment);
-        final KeyStore trustStore = CertParsingUtils.readKeyStore(path, keyStoreType, keyStorePassword.getChars());
-        final List<CertificateInfo> certificates = new ArrayList<>();
-        final Enumeration<String> aliases = trustStore.aliases();
-        while (aliases.hasMoreElements()) {
-            String alias = aliases.nextElement();
-            final Certificate[] chain = trustStore.getCertificateChain(alias);
-            if (chain == null) {
-                continue;
-            }
-            for (int i = 0; i < chain.length; i++) {
-                final Certificate certificate = chain[i];
-                if (certificate instanceof X509Certificate) {
-                    certificates.add(new CertificateInfo(keyStorePath, keyStoreType, alias, i == 0, (X509Certificate) certificate));
+        // Do not return certificates from the PKCS#11 token, as it is also the JRE default store
```

Comment:
@jaymode I wanted to point out this change ( to not return the certificates of the PKCS#11 token in the get certificates API responses ) since I introduced it after your LGTM.

Reply:
hmm, this seems odd since the idea is to be able to use this API to alert about certificate expiration. Wouldn't we want this if the cert in the PKCS11 token is going to expire?